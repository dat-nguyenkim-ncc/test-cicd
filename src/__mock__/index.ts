import { TagGroupType } from './../types/index.d'
import {
  CompanyBusiness,
  CompanyDetail,
  CompanyFinancials,
  CompanyOverview,
  CompanyPeople,
  MappingSummary,
  TagData,
  TagDataParent,
} from '../types'
import { EnumAttachmentType, EnumCompanyTypeSector } from '../types/enums'

const getRandomId = (): string => {
  return (~~(Math.random() * 999999) + new Date().valueOf()).toString()
}

const companyTypes = Object.values(EnumCompanyTypeSector)
const suffix = ['Restaurant', 'Limited', 'UK', 'US', 'Bank']
export const mockCompanyId = getRandomId()
export const companyDetails = (): CompanyDetail => ({
  companyId: mockCompanyId,
  companyName: `Monzo ${suffix[~~(Math.random() * suffix.length)]}`,
  countryCode: 'GB',
  countryName: 'United Kingdom',
  url: 'https://monzo.com',
  companyTypes: [companyTypes[~~(Math.random() * companyTypes.length)]],
})

export const MockMonzoDetails = companyDetails()

export const mappingSummary = [
  {
    title: 'CATEGORIES',
    items: [
      { label: 'Fintech', parent: [] },
      { label: 'Retail Accounts' },
      { label: 'Accounts & Savings' },
      { label: 'Digital Banking' },
    ],
    type: EnumCompanyTypeSector.FIN,
  },
  {
    title: 'SECTORS',
    items: [{ label: 'No sector mapped', parent: [] }],
    type: EnumCompanyTypeSector.FIN,
  },
] as MappingSummary[]

export const companySources = [
  {
    company: MockMonzoDetails,
    source: {
      label: 'Source: XPTO',
      default: true,
    },
  },
  {
    company: companyDetails(),
    source: {
      label: 'Manual',
    },
  },
]

export const companyOverview = (): CompanyOverview => ({
  id: getRandomId(),
  source: 'CRB',
  sources: ['CRB', 'DR', 'MANUAL'],
  companyName: `Monzo ${suffix[~~(Math.random() * suffix.length)]}`,
  companyLocation: [
    {
      id: '123456',
      address: 'Broadwalk House<br>5 Appold Street<br>London<br>EC2A 2AG',
      postalCode: '94103',
      location: {
        region: 'EMEA',
        city: 'San Francisco',
        country: 'United States',
      },
      isHeadQuarter: true,
      selfDeclared: true,
      expandStatus: 'Following',
      source: 'bcg',
    },
  ],
  description: 'Monzo is a fintech challenger bank.',
  companyType: 'Startup',
  foundedYear: 2012,
  status: 'The company status',
  expandStatus: 'Following',
  expandStatusId: '1',
  closedDate: new Date(2012, 11, 1).toISOString(),
  ftse: 1000,
  logoUrl: 'https://monzo.com/logo.svg',
  contactEmail: 'monzo@monzo.com',
  phoneNumber: '0123456789',
  tags: null,
  otherNames: null,
  lastFundingType: null,
  logo_bucket_url: '',
  hashed_image: '',
  attachments: [
    {
      name: 'File nam',
      description: 'Broadwalk House<br>5 Appold Street<br>London<br>EC2A 2AG',
      type: EnumAttachmentType.FCT_PROFILE,
      url_attachment: '',
      expandStatus: 'Following',
      selfDeclared: true,
      date_created: '2021-03-30 10:43:33',
    },
  ],
  aliases: [
    {
      alias_id: '200354bcg1',
      company_alias: 'name-edit',
      expand_status_id: 1,
      selfDeclared: true,
      source: 'bcg',
    },
    {
      alias_id: '200354bcg2',
      company_alias: 'test1',
      expand_status_id: 1,
      selfDeclared: true,
      source: 'bcg',
    },
  ],
})

export const companyFinancials = (): CompanyFinancials => ({
  valuation: {
    value: 23123123,
    currency: 'USD',
  },
  fundingTotal: {
    value: 2312312312,
    currency: 'USD',
  },
  companyStage: 'Series E+',
  lastFundingDate: {
    value: 1231233,
    currency: 'USD',
  },
  equityFundingTotal: {
    value: 1231233,
    currency: 'USD',
  },
  leadInvestor: 'Mussum Cassilds-Cepacol',
  lastFundingAmount: {
    value: 123123123,
    currency: 'USD',
  },
  fundingRounds: [
    {
      id: '1',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
        other:
          'Goodwater Capital, VC Orange, Corporate Passion Capital, VC Reference Capital (formerly Genevest), VC Stripe, Tech Thrive Capital, VC Vanderbilt University, Other Y Combinator, Accelerator',
      },
    },
    {
      id: '2',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
      },
    },
    {
      id: '3',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
      },
    },
  ],
  isExternalViewDetail: true,
})

export const companyBusiness = (): CompanyBusiness => ({
  productsServices: ['Offers a current account and debit card '],
  businessRevenue: ['Service type: B2C'],
  targetClients: ['Retail customers'],
  partnerships: [
    '**Deliveroo:** Partnered with Deliveroo in Sep-2017 to allow users to split the bill on orders via its Monzo.me money request feature',
  ],
  vision: ['string'],
  keyMetrics: ['string'],
  differentiators: ['string'],
})

export const companyAcquisitions = () => []
export const companyIpos = () => []
export const companyPeople = (): CompanyPeople => ({ companyId: 145803 })

const tags = [
  'Capital Markets',
  'Corporate Banking',
  'Retail Banking',
  'SME Banking',
  'Support',
  'Technology',
  'Wealth Management',
]

const getRandomValueLabel = (index?: number) => {
  const tag = index ? tags[index] : tags[~~(Math.random() * tags.length)]
  return {
    label: tag,
    value: tag.toLowerCase().replace(' ', '_'),
  }
}

const getRandomBoolean = () => {
  return Math.random() * 10 > 5
}

export const TagMock = (
  index?: number,
  level: number = 0,
  parent: TagDataParent[] = []
): TagData => {
  const { value, label } = getRandomValueLabel(index)
  const id = getRandomId()
  const rowId = getRandomId()
  const isPriority = getRandomBoolean()
  return {
    id,
    rowId,
    value,
    label,
    isPriority,
    endpoint: level > 0 ? Math.random() > 0.5 : false,
    parent,
    description: 'Mock',
    children:
      level >= 5
        ? undefined
        : new Array(Math.max(3, ~~(Math.random() * tags.length)))
            .fill(null)
            .map((_a, index) =>
              TagMock(index, level + 1, [...parent, { id, rowId, label, value, isPriority }])
            ),
  }
}

export const ListRandomTags = (): TagData[] =>
  new Array(tags.length).fill(null).map((_a, index) => TagMock(index, 0))

const labels = [
  'Insurance Value Chain 1',
  'Insure Tech Tag',
  'Revenue model',
  'P&C Commercial Customers',
  'Product',
  'RegTech',
  'Regulation',
  'Report',
]

export const TagGroupMock = (): TagGroupType[] =>
  new Array(labels.length * 2).fill(null).map((_a, index) => {
    const id = getRandomId()
    const rowId = getRandomId()
    const label = labels[index % labels.length]
    const value = label.toLowerCase().split(' ').join('_')
    const isPriority = getRandomBoolean()
    return {
      id,
      rowId,
      label,
      value,
      isPriority,
      parent: [],
      children: new Array(Math.max(1, ~~(Math.random() * labels.length * 2)))
        .fill(null)
        .map((_b, index) => {
          const labelChild = labels[index % labels.length]
          return {
            label: labelChild,
            id: getRandomId(),
            rowId: getRandomId(),
            parent: [{ id, label, value }],
            isPriority,
            value: labelChild.toLowerCase().split(' ').join('_'),
          }
        }),
    }
  })
