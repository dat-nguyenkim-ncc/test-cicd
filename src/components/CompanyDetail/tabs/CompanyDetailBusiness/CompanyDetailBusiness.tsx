import React from 'react'
import { Box } from 'theme-ui'
import { CompanyPeople } from '../../../../types'
import { Paragraph } from '../../../primitives'
import {
  getProfile,
  getProfileType,
  GET_FINANCIAL_SERVICES_LICENSES,
} from '../../../../pages/CompanyForm/graphql'
import { ProfileType } from '../../../../pages/CompanyForm/helpers'
import { useQuery } from '@apollo/client'
import { Updating } from '../../..'
import { FinancialServiceOverview, ProfileEditType, ProfileGroupItem } from '../../../ProfileForm'
import { EnumExpandStatusId } from '../../../../types/enums'
import {
  FinanceServiceLicense,
  FinanceServiceLicenseType,
} from '../../../ProfileForm/FinanceServicesLicenses'

export type CompanyDetailBusinessProps = {
  data: CompanyPeople
}

const CompanyDetailBusiness = ({ data }: CompanyDetailBusinessProps) => {
  // GRAPHQL
  const { data: profileType, loading } = useQuery<{
    getProfileType: Array<ProfileType>
  }>(getProfileType)
  const { data: profiles, loading: profileLoading } = useQuery<{
    getProfile: Array<ProfileEditType>
  }>(getProfile, {
    fetchPolicy: 'network-only',
    variables: {
      id: +data.companyId,
    },
  })
  const { data: financeLicenses, loading: financeLicensesLoading } = useQuery<{
    getCompanyFinancialServicesLicenses: Array<FinanceServiceLicense>
  }>(GET_FINANCIAL_SERVICES_LICENSES, {
    fetchPolicy: 'network-only',
    variables: {
      companyId: +data.companyId,
    },
  })

  const financeLicenseData = (financeLicenses?.getCompanyFinancialServicesLicenses || []).filter(
    ({ fctStatusId }) => fctStatusId === +EnumExpandStatusId.FOLLOWING
  )

  return loading || profileLoading || financeLicensesLoading ? (
    <Updating loading />
  ) : (
    <Box mt={6}>
      {profileType
        ? profileType.getProfileType.map(
            ({ profile_type_id, profile_type_name, options, group }, index) => {
              const state = profiles?.getProfile.filter(
                p =>
                  (p.profile_type_id === profile_type_id ||
                    group?.some(g => g.profile_type_id === p.profile_type_id)) &&
                  p.expand_status_id === EnumExpandStatusId.FOLLOWING
              )

              const profileTypeIds = (group || []).map(({ profile_type_id }) => profile_type_id)

              const financeServiceLicenseType =
                // @ts-ignore
                profileTypeIds.includes('0') && profileTypeIds.includes(null)

              return (
                <Box sx={{ mb: 5 }} key={index}>
                  {state?.length ||
                  (financeServiceLicenseType && group && !!financeLicenseData.length) ? (
                    <>
                      <Paragraph sx={{ mb: 3 }} bold>
                        {profile_type_name}
                      </Paragraph>
                      {financeServiceLicenseType && group && !!financeLicenseData.length ? (
                        <>
                          <FinancialServiceOverview
                            group={
                              group.map(item => ({
                                ...item,
                                field:
                                  item.profile_type_id === '0'
                                    ? 'license_jurisdiction'
                                    : 'license_type',
                              })) as FinanceServiceLicenseType[]
                            }
                            oldState={financeLicenseData}
                          />
                        </>
                      ) : group && state ? (
                        <>
                          <ProfileGroupItem sx={{ my: 1 }} group={group} state={state} />
                        </>
                      ) : (
                        <>
                          <Box
                            sx={{
                              bg: 'gray03',
                              my: 1,
                              p: 4,
                              borderRadius: '10px',
                              width: '100%',
                              border: '1px solid',
                              borderColor: 'gray01',
                            }}
                          >
                            {(state || []).map((item, i) => {
                              return (
                                <Paragraph sx={{ mt: i > 0 ? 1 : 0 }} key={i}>
                                  {item.profile_value}
                                </Paragraph>
                              )
                            })}
                          </Box>
                        </>
                      )}
                    </>
                  ) : undefined}
                </Box>
              )
            }
          )
        : undefined}
    </Box>
  )
}

export default CompanyDetailBusiness
