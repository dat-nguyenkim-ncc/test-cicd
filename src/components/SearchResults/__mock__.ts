export const data = {
  external: [
    {
      //replace property id by company_id
      //TODO: change all records
      company_id: '7b4545e6-a34d-47c2-853e-ff9eb40c18ef',
      source: 'CRB',
      companyDetails: {
        companyId: '9185a077-c260-ac40-2678-fef9e645591b',
        companyName: 'Monzo',
        url: 'https://monzo.com/',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: 'a67f6ebd-c2e4-4c15-8657-4d40cc7626f1',
      source: 'DR',
      companyDetails: {
        companyId: '74473',
        companyName: 'Monzo',
        url: 'https://monzo.com',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: '89c20324-1390-4ea2-acbc-7cc05a98db59',
      source: 'CRB',
      companyDetails: {
        companyId: '33164743-a09a-45cb-a160-a54ace4dc3da',
        companyName: 'Monzon Biomed',
        url: 'http://www.monzonbiomed.com/',
        countryName: 'Spain',
        countryCode: 'ES',
      },
    },
    {
      id: '807a6c17-a4ff-4b74-b9ef-fee94364f451',
      source: 'CRB',
      companyDetails: {
        companyId: '456754eb-67c6-98db-780a-327db4c05c8f',
        companyName: 'monzoom.com',
        url: 'https://www.monzoom.com',
        countryName: 'Austria',
        countryCode: 'AT',
      },
    },
    {
      id: '0511b523-c25b-443b-932f-4d501d5c0cd4',
      source: 'CRB',
      companyDetails: {
        companyId: '09c52685-0f60-4a6f-a5fc-5c51ae5bec3a',
        companyName: 'Monzoon Networks',
        url: 'http://hotspot.monzoon.net',
        countryName: 'Switzerland',
        countryCode: 'CH',
      },
    },
  ],
  internal: [
    {
      //replace property id by company_id
      //TODO: change all records
      company_id: '6',
      source: 'MANUAL',
      companyDetails: {
        companyId: '6',
        companyName: 'MonzoBCG',
        url: 'www.MonzoBCG.com',
        countryName: 'England',
        countryCode: null,
      },
    },
    {
      id: '127237',
      source: 'DR',
      companyDetails: {
        companyId: '127237',
        companyName: 'Monzo',
        url: 'https://monzo.com',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: '141132',
      source: 'MANUAL',
      companyDetails: {
        companyId: '141132',
        companyName: 'Monzo',
        url: 'https://monzo.com/',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
  ],
}
