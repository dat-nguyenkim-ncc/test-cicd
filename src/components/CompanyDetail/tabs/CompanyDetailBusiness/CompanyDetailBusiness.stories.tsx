import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import CompanyDetailBusiness, { CompanyDetailBusinessProps } from './CompanyDetailBusiness'
import { companyPeople } from '../../../../__mock__'
import { Section } from '../../../primitives'

export default {
  title: 'Company Details/CompanyDetailBusiness',
  component: CompanyDetailBusiness,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta

const Template: Story<CompanyDetailBusinessProps> = args => (
  <Section sx={{ m: 6, mx: 'auto' }}>
    <CompanyDetailBusiness {...args} />
  </Section>
)

export const Default = Template.bind({})
Default.args = {
  data: companyPeople(),
} as CompanyDetailBusinessProps
