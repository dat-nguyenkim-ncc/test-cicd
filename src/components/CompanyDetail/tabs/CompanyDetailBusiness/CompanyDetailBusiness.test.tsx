import * as React from 'react'
import { shallow } from 'enzyme'
import CompanyDetailBusiness from './CompanyDetailBusiness'
import { companyPeople } from '../../../../__mock__'
import { MockedProvider } from '@apollo/client/testing'

describe('CompanyDetailBusiness', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <MockedProvider addTypename={false}>
        <CompanyDetailBusiness data={companyPeople()} />
      </MockedProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
