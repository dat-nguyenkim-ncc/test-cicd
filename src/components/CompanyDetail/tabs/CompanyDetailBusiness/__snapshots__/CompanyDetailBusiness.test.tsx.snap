// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDetailBusiness matches snapshot 1`] = `ShallowWrapper {}`;
